src/App.js

```diff
@@ -43,19 +48,38 @@ function App() {
     console.log(cartProducts)
   };
 
-  const onAddToFavorite = (obj) => {
-    setFavorites(prev => {
-      // Check if the item already exists in favorites
-      if (!prev.some(item => item.id === obj.id)) {
-        // If it doesn't exist, add it to the favorites array
-        return [...prev, obj];
+  const onAddToFavorite = async (obj) => {
+    console.log(obj)
+    console.log()
+    if (obj.name) {
+      try {
+        if ((favorites).find(item => Number(item.id) === Number(obj.id))) {
+          axios.delete(`https://64707a693de51400f7245522.mockapi.io/favorites/${obj.id}`);
```

Comment:
method is called "onAddToFavorite" but you "delete" here

Reply:
This function deletes duplicates only